### PR TITLE
S8_implement_create_task_post_api

### PR DESCRIPTION
**Pull request by denys-rybachok**
Added endpoint for create task, added validation.

### CODE REVIEW

File: constants/task.enum.js [hunk 1]
```diff
@@ -6,7 +6,13 @@ export const PriorityEnum = {
 };
 
 export const StatusEnum = {
-  NEW: 1,
-  IN_PROGRESS: 2,
-  COMPLETED: 3
+  IN_PROGRESS: 1,
+  COMPLETED: 2
```
Comment (VYaniv) [inline constants/task.enum.js:10]: What about new?

File: routes/task.routes.js [hunk 2]
```diff
@@ -1,8 +1,11 @@
 import express from 'express';
-import {getTasks} from '../controllers/Task.controller.js';
+import {getTasks, createTask} from '../controllers/Task.controller.js';
+import {createTaskValidation} from '../validation/task.validation.js';
+import handleValidationErrorsUtil from '../utils/handleValidationErrors.util.js';
 
 const router = express.Router();
 
 router.get('/', getTasks);
```
Comment (VYaniv) [inline routes/task.routes.js]: better to have `/task` for get all. And create constants for routes.

File: routes/task.routes.js [hunk 3]
```diff
@@ -1,8 +1,11 @@
...
+router.post('/create', createTaskValidation, handleValidationErrorsUtil, createTask);
```
Comment (VYaniv) [inline routes/task.routes.js]: the same hare

File: services/task.service.js [hunk 4]
```diff
@@ -8,3 +8,13 @@ export const getAllTasks = async () => {
...
   }
 };
+
+export const createTask = async (data) => {
+  try {
+    const newTask = new Task({...data});
+    const savedTask = await newTask.save();
+    return savedTask;
+  } catch (error) {
+    throw new Error('Error creating tasks');
```
Comment (VYaniv) [inline services/task.service.js]: Error constants.

File: services/task.service.js [hunk 5]
```diff
@@ -8,3 +8,13 @@ export const getAllTasks = async () => {
     throw new Error('Error fetching tasks');
   }
 };
+
+export const createTask = async (data) => {
+  try {
+    const newTask = new Task({...data});
+    const savedTask = await newTask.save();
+    return savedTask;
```
Comment (VYaniv) [inline services/task.service.js:19]: Also cane empty line before return. It looks more readable.

File: constants/task.enum.js (hunk 1 above)
Comment (denys-rybachok) [reply constants/task.enum.js:10]: Ok, I will use 'NEW' instead of 'IN_PROGRESS'.

File: services/task.service.js (hunk 4 above)
Comment (denys-rybachok) [reply services/task.service.js]: Ok.

File: services/task.service.js (hunk 5 above)
Comment (denys-rybachok) [reply services/task.service.js:19]: Ok, thanks.

File: routes/task.routes.js (hunk 2 above)
Comment (denys-rybachok) [reply routes/task.routes.js]: Now my routes are: 
http://localhost:4444/task - for get all
http://localhost:4444/task/create - for post
You suggest to use one common http://localhost:4444/task for get and post? 

create constants for routes. - ok. Thanks.

Comment (VYaniv) [reply routes/task.routes.js]: Yes

**Review by VYaniv — APPROVED** [review]

**Pull request merged**